From a21878e5dca2bc519ec1d970c34e8f8280b73854 Mon Sep 17 00:00:00 2001
From: Ram Kumar D <quic_ramd@quicinc.com>
Date: Fri, 30 Aug 2024 16:21:20 +0530
Subject: [PATCH] drivers: spi: enable resource before geni fw load

geni fw load expects se clk to be enabled. Currently,
it is getting loaded since the clk already got enabled
by the bootloader.

So, this patch enables se clk before the geni fw load.

Change-Id: If303c693db734226d921acdbdee9db4b8df14ace
Signed-off-by: Ram Kumar D <quic_ramd@quicinc.com>
---
 drivers/spi/spi-geni-qcom.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/spi/spi-geni-qcom.c b/drivers/spi/spi-geni-qcom.c
index f957139bdd7d..e76d16a0f292 100644
--- a/drivers/spi/spi-geni-qcom.c
+++ b/drivers/spi/spi-geni-qcom.c
@@ -1087,10 +1087,6 @@ static int spi_geni_probe(struct platform_device *pdev)
 		return ret;
 	}
 
-#ifdef CONFIG_QCOM_GENI_SE_FW_LOAD
-	geni_se_fw_load(&mas->se, QUPV3_SE_SPI);
-#endif /* CONFIG_QCOM_GENI_SE_FW_LOAD */
-
 	spi->bus_num = -1;
 	spi->dev.of_node = dev->of_node;
 	spi->mode_bits = SPI_CPOL | SPI_CPHA | SPI_LOOP | SPI_CS_HIGH;
@@ -1130,6 +1126,14 @@ static int spi_geni_probe(struct platform_device *pdev)
 	if (ret)
 		goto spi_geni_probe_runtime_disable;
 
+#ifdef CONFIG_QCOM_GENI_SE_FW_LOAD
+	ret = geni_se_resources_on(&mas->se);
+	if (ret)
+		return ret;
+
+	geni_se_fw_load(&mas->se, QUPV3_SE_SPI);
+#endif /* CONFIG_QCOM_GENI_SE_FW_LOAD */
+
 	ret = spi_geni_init(mas);
 	if (ret)
 		goto spi_geni_probe_runtime_disable;
-- 
2.34.1

